ls1intum/Artemis
# `Quiz exercises`: Fix batch quiz editing when batch is added

### Checklist
#### General


#### Client

### Motivation and Context



An instructor can still visit the quiz edit page even though the batched quiz has batches already by clicking the quiz id on the exercises page or by visiting directly the edit page url.

### Description


- Disable the quiz id link to the edit page when the batched quiz has batches.
- Show an unauthorized popup message and disable the page if the quiz is not editable and the edit page link is visited.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Course

1. Log in to Artemis
2. Navigate to Course Administration
3. Select the Course
4. Navigate to Exercises
5. Create a new Quiz Exercise
6. Set Batch Mode to **Batched**
7. Add 1 Question of any type
8. Click Save
9. Click Back
10. Click Edit on the newly created quiz
11. Verify that the quiz is still editable
12. Click Back
13. Click **+ Add Batch**
14. Verify that the edit button is disabled and the quiz id is not clickable
15. Verify that when the edit link is visited, the page shows an unauthorized popup message and the page does not show anything.
16. Verify that importing the quiz exercise works.

Note: See the screencast for the detailed steps for testing.

### Review Progress






#### Code Review
#### Manual Tests

### Test Coverage



| Class/File | Branch | Line |
|------------|-------:|-----:|
| quiz-exercise-detail.component.ts | 79.79% | 94.66% |
| quiz-exercise.component.ts | 58.1% | 88.52% |
| quiz-manage-util.service.ts | 100% | 100% |

### Screenshots



https://user-images.githubusercontent.com/10885503/210074261-88cb4a38-3afd-45b7-9060-7ce7c24b8cf9.mov
